tests/failpoints_cases/test_snap.rs

```diff
@@ -231,13 +231,19 @@ fn test_generate_snapshot() {
 }
 
 fn must_empty_dir(path: String) {
-    for _ in 0..200 {
-        let snap_dir = fs::read_dir(&path).unwrap();
-        if snap_dir.count() > 0 {
-            thread::sleep(Duration::from_millis(10));
-            continue;
+    for i in 0..500 {
+        thread::sleep(Duration::from_millis(10));
+        let entries = fs::read_dir(&path)
+            .and_then(|dir| dir.collect::<io::Result<Vec<_>>>())
+            .unwrap();
+        if entries.is_empty() {
+            return;
+        }
+        if i == 499 {
```

Comment:
This is weird. Why not put panic outside the loop?

Reply:
+1